src/tree.rs

```diff
@@ -258,9 +257,9 @@ impl<T: PartialEq + TreeHash + Clone + Encode + Decode> Tree<T> {
             }
             (Self::PackedLeaf(l1), Self::PackedLeaf(l2)) if depth == 0 => {
                 let mut equal = true;
-                for i in 0..l2.values.len() {
-                    let v2 = &l2.values[i];
-                    match l1.values.get(i) {
+                for i in 0..l2.length() {
+                    let v2 = l2.get(i).expect("Is always in-bounds"); // FIXME
```

Comment:
Another panic here, we should return an error instead (I know the previous behaviour was panicky.. but would be nice to improve it).